awswrangler/s3/_read_text_core.py

```diff
@@ -0,0 +1,114 @@
+"""Amazon S3 Read Core Module (PRIVATE)."""
+import logging
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union
+
+import boto3
+import botocore.exceptions
+import pandas as pd
+import pandas.io.parsers
+from pandas.io.common import infer_compression
+
+from awswrangler import _utils, exceptions
+from awswrangler.s3._fs import open_s3_object
+from awswrangler.s3._read import _apply_partitions
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _get_read_details(path: str, pandas_kwargs: Dict[str, Any]) -> Tuple[str, Optional[str], Optional[str]]:
+    if pandas_kwargs.get("compression", "infer") == "infer":
+        pandas_kwargs["compression"] = infer_compression(path, compression="infer")
+    mode: str = "r" if pandas_kwargs.get("compression") is None else "rb"
+    encoding: Optional[str] = pandas_kwargs.get("encoding", "utf-8")
+    newline: Optional[str] = pandas_kwargs.get("lineterminator", None)
+    return mode, encoding, newline
+
+
+def _read_text_chunked(
+    path: str,
+    chunksize: int,
+    parser_func: Callable[..., pd.DataFrame],
+    path_root: Optional[str],
+    boto3_session: boto3.Session,
+    pandas_kwargs: Dict[str, Any],
+    s3_additional_kwargs: Optional[Dict[str, str]],
+    dataset: bool,
+    use_threads: Union[bool, int],
+    version_id: Optional[str] = None,
+) -> Iterator[pd.DataFrame]:
+    mode, encoding, newline = _get_read_details(path=path, pandas_kwargs=pandas_kwargs)
+    with open_s3_object(
+        path=path,
+        version_id=version_id,
+        mode=mode,
+        s3_block_size=10 * 1024 * 1024,  # 10 MB (10 * 2**20)
```

Comment:
So was the `READER_ROW_BATCH_SIZE` chosen to be 10Mb in order to match this s3_block_size or is it just a coincidence?

Reply:
It was mostly about matching the batch size used for the S3 Parquet reader, which also uses 10MiB. And I saw that the S3 block size was also 10 MiB, so it seemed like a good number to use. I'm not sure if there's any more specific guidance for how to choose this number.